.circleci/config.yml

```diff
@@ -5,194 +5,98 @@
-            # https://discuss.circleci.com/t/disable-autodetection-of-project-or-application-of-python-venv/235/10
-            cd && rm -rf ~/.pyenv && rm -rf ~/virtualenvs
+jobs:
 
-            # We need to remove conflicting texlive packages.
-            sudo -E apt-get -yq remove texlive-binaries --purge
-      - run:
-          name: Install packages for make -C doc check
-          command: |
-            # Installing required packages for `make -C doc check command` to work.
-            sudo -E apt-get -yq update
-            sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install dvipng texlive-latex-base texlive-latex-extra
+  build_docs:
+    
+    docker:
+      - image: circleci/python:3.8
 
-  restore_from_cache:
-    description: "Restores the cache of previously built docs & python packages if present"
     steps:
-      - run:
-          name: Generate cache keys from today's date for built docs & week number for python packages
-          command: |
-            date +%F > today
-            date +%U > week_num
       - restore_cache:
           keys:
-            - v1-packages+datasets-{{ checksum "week_num" }}-{{ checksum ".circleci/packages-cache-timestamp" }}
-            - v1-packages+datasets-U3h5YwdTXfPsjYsVouLcVkFBnD0wYM_jIjjA+pc_eqM=
-            - v1-packages+datasets
-      - restore_cache:
-         key: v1-docs-{{ .Branch }}-{{ checksum "today" }}-{{ checksum ".circleci/docs-cache-timestamp" }}
-
-  cache_aware_conda_setup:
-    description: "Downloads & installs conda if not restored by cache"
-    steps:
+              - source-cache
+      - checkout
       - run:
-          name: Download & install conda if absent
+          name: Complete checkout
           command: |
-            if
-              ls $HOME/miniconda3/bin | grep conda -q
-            then
-              echo "(Mini)Conda already present from the cache."
-            else
-              wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
-              chmod +x ~/miniconda.sh && ~/miniconda.sh -b
-            fi
+              if ! git remote -v | grep upstream; then
+                  git remote add upstream git://github.com/nilearn/nilearn.git
+              fi
+              git fetch upstream
+      - save_cache:
+          key: source-cache
+          paths:
+              - ".git"
       - run:
-          name: Setup conda path in env variables
+          name: Merge with upstream
           command: |
-            echo 'export PATH="$HOME/miniconda3/bin:$PATH"'  >> $BASH_ENV
+              echo $(git log -1 --pretty=%B) | tee gitlog.txt
+              echo "gitlog.txt = $(cat gitlog.txt)"
+              echo ${CI_PULL_REQUEST//*pull\//} | tee merge.txt
+              if [[ $(cat merge.txt) != "" ]]; then
+                  echo "Merging $(cat merge.txt)";
+                  git pull --ff-only upstream "refs/pull/$(cat merge.txt)/merge";
```

Comment:
is it always a fast-forward?

Reply:
Yes (unless there is a conflict of course) because we use `refs/pull/<number>/merge` which references the merge commit between `refs/pull/<number>/head` and the destination branch.